verilog/CST/package_test.cc

```diff
@@ -36,59 +36,250 @@
+
+        {{kTag, "package p; \n endpackage"},
+        " let Peace = Love;\n",
+        {kTag, "package p; \n endpackage"},
+        " let Five() = Two + Two + One;\n",
+        " let Min(a,b) = (a < b) ? a : b;\n",
+        " let Max(a,b=1) = (a > b) ? a : b;\n",
+        {kTag, "package p; \n endpackage"},
+        " let Max(untyped a, bit b=1) = (a > b) ? a : b;\n"},
+
+        {"localparam real foo = 3.14;\n",
+        "localparam shortreal foo = 159.265;\n",
+        "localparam realtime foo = 358.979ns;\n",
+        {kTag, "package p; \n endpackage"},
+        "  localparam real foo = 323.846;\n"},
+
+        {{kTag, "package p; \n endpackage"},
+        " import $unit::arnold;\n",
+        " import $unit::*;\n",
+        {kTag, "package p; \n endpackage"}},
+
+        {{kTag, "package p; \n endpackage"},
+        " export bar::baz;\n",
+        " export bar::*;\n",
+        {kTag, "package p; \n endpackage"}},
+         {{kTag, "package p; \n endpackage"},
+        " parameter reg[BITS:0] MR0 = '0;\n"},
+
+        {{kTag, "package p; \n endpackage"},
+        " bind scope_x type_y z (.*);\n",
+        " bind scope_x type_y z1(.*), z2(.*);\n",
+        " bind module_scope : inst_x type_y inst_z(.*);\n",
+        {kTag, "package p; \n endpackage"}},
+
+        {{kTag, "package p; \n endpackage"},
+        "`ifndef DEBUGGER\n",
+        "`endif\n",
+        "  int num_packets;\n",
+        "`ifdef DEBUGGER\n",
+        "`endif\n",
+        "  int router_size;\n"},
+
+        {{kTag, "package p; \n endpackage"},
+        "  int num_packets;\n",
+        "`ifdef DEBUGGER\n",
+        "`elsif BORED\n",
+        "`else\n",
+        "  string source_name;\n",
+        "  string dest_name;\n",
+        "`endif\n",
+        "  int router_size;\n",
+        {kTag, "package p; \n endpackage"}},
+
+         {{kTag, "package p; \n endpackage"},
+          " virtual a_if b_if;\n",
+          "virtual a_if b_if, c_if;\n",
+          {kTag, "package p; \n endpackage"},
+          "  virtual a_if b_if;\n",
+          {kTag, "package p; \n endpackage"}},
+
+          {{kTag, "package p; \n endpackage"},
+          "`ifdef DEBUGGER\n",
+          "`endif\n",
+          {kTag, "package p; \n endpackage"},
+          "`ifdef DEBUGGER\n",
+          "`ifdef VERBOSE\n"  ,
+          "`endif\n",
+          "`endif\n",
+          {kTag, "package p; \n endpackage"}}
+
+  };
+  for( const auto & test : testcases)
+  {
+    const absl::string_view code(test.code);
+    VerilogAnalyzer analyzer(code, "test-file");
+    const auto code_copy = analyzer.Data().Contents();
+    ASSERT_OK(analyzer.Analyze()) << "failed on:\n" << code;
+    const auto& root = analyzer.Data().SyntaxTree();
+
+    const auto decls = FindAllPackageDeclarations(*ABSL_DIE_IF_NULL(root));
+
+    std::ostringstream diffs;
+    EXPECT_TRUE(test.ExactMatchFindings(decls, code_copy, &diffs))
+        << "failed on:\n"
+        << code << "\ndiffs:\n"
+        << diffs.str();
+  }
```

Comment:
See this recent commit:
https://github.com/google/verible/commit/b35874a4f62f4f479973c15adde60961b70ac139

This provides an easier way to drive these tests without so much copy-pasting.

Reply:
i think i didnt pull the latest from when i changed it in #400